guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+# Eve JavaScript/TypeScript DSL
+
+The forthcoming v0.3 release of Eve supports a domain specific langauge (DSL) that interacts with the Eve server, allowing you to intermix Javascript and Eve code. There are several benefits to using the DSL:
```

Comment:
there's no server at the moment, so probably worth mentioning this is client-side only for now